src/common/telemetry/lib/Telemetry.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef TELEMETRY_HPP
+#define TELEMETRY_HPP
+
+#include <memory>
+#include <mutex>
+#include <parson.h>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <Keys.h>
+#include <LogManager.hpp>
+#include <Logging.h>
+
+#include "ParameterSets.hpp"
+
+namespace Telemetry
+{
+
+    class TelemetryManager
+    {
+    public:
+        static const int CONFIG_DEFAULT_TEARDOWN_TIME = 5; // seconds
+
+        // Singleton pattern - get the global instance
+        static TelemetryManager& GetInstance();
+
+        // Delete copy constructor and assignment operator to enforce singleton
+        TelemetryManager(const TelemetryManager&) = delete;
+        TelemetryManager& operator=(const TelemetryManager&) = delete;
+
+        // Initialize telemetry system
+        bool Initialize(bool enableDebug = false, int teardownTime = CONFIG_DEFAULT_TEARDOWN_TIME);
+
+        // Check if telemetry is initialized
+        bool IsInitialized() const;
```

Comment:
> bool IsInitialized() const; [](http://example.com/codeflow?start=7&length=28)

This one can be a private function and be called by ProcessJsonFile. 

The client does not need to see GetInstance, Initialize, and Shutdown. All these can be automatically called internally by ProcessJsonFile.